### ISSUE TITLE
[Chips] Support styled text fields for Input Chips

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#4178]
## Story
As a client using Material Input Chips, I want a way to style a UITextField (or similar) similarly to Material Text Fields so that chip entry looks similar to text entry in my application.

## Definitions of Done
* The text field will look and behave like Material Text Fields using the Outlined and Filled styles.
* The text field can support a single row of chips and can scroll.
* The text field can support multiple rows of chips and not scroll.

## References
* [Material Guidelines for Input Chips](https://material.io/design/components/chips.html#input-chips)
* [Material Guidelines for Text Fields](https://material.io/design/components/text-fields.html#usage)

**Comment by bhamiltoncx** [comment]
Please also support shrinking the placeholder text like a Material Text Field.

**Comment by romoore** [comment]
@bhamiltoncx What do you mean by shrinking the placeholder text?  I think floating placeholders do a bit of shrinking, but I don't think we support it in general. Is the floating placeholder behavior what you're referring to?

**Comment by bhamiltoncx** [comment]
@romoore In the top example here:

https://material.io/develop/ios/components/textfields/

the "Name" placeholder text shrinks and turns blue when the text field is selected.

Currently, the placeholder text in a Chips field disappears when the field is selected.

I would like Chips fields to behave the same as text fields when selected.

**Comment by romoore** [comment]
Got it, thanks for clarifying.  I believe that behavior is already part of the Text Field design. I thought you wanted *any* placeholder to be shrinkable.  I agree that the text field for Input chips should behave similarly to Material Text Fields. Is there a better way to phrase the **Definition of Done** above to capture the requirement more clearly?

**Comment by bhamiltoncx** [comment]
If you think the second bullet point covers all aspects of Text Fields, we don't need any change. :) I just wanted to make sure we explicitly handled that particular feature.

**Comment by romoore** [comment]
Yeah, after a discussion with @willlarche today, I'm going to take a look at extending Text Fields to support Chips. I think given the requirement to mix text and chips throughout the content, the current ChipField implementation may not be sufficient.

**Comment by mohammadcazig** [comment]
Can we also verify with Design that having textfields with Filled our outlined style in inputChip is acceptable? I have not seen any specs nor any place where we currently have such behavior.

**Comment by romoore** [comment]
Looks like it should be possible using attributed text. 

**General Idea**
1. Create a Chip with the appropriate text and determine its intrinsic size
1. Create a clear image of the same size as the Chip's bounds
1. Replace the chip's original text with an attributed string containing an `NSTextAttachment`. The `image` will be the generated clear image as a placeholder for the Chip.
1. Position the Chip view over the generated image, making it appear that the Chip is inline with the text

**Caveats/Concerns**
1. This can only be achieved using a (subclass of) `UITextView`. `UITextField` does not seem to support `NSTextAttachment` attributes.
1. This permits "interleaving" of the text and Chips, but generating a sensible copy of the "original" text will require special handling.
1.  The original `MDCChipField` had special handling of return key presses (`CR` and/of `LF` characters). Very careful consideration needs to be taken with respect to newlines and carriage returns because UITextView has no analogue for `-textFieldShouldReturn:`. As a result, pasting a newline or carriage return should be considered when searching for chip delimiters.

**Comment by romoore** [comment]
@mohammadcazig  After talking to @willlarche about this last week, I believe the current mocks show "Full width" text fields, which are not the recommended styles for text input (at this time). I have requested feedback/recommendations from the appropriate designers, but I think it's reasonable to move forward with this story.  Since Material currently only recommends Outlined/Filled styles for text input, and text input is required for Input Chips, it seems natural that Chips would use a recommended text input style.

**Comment by mpasumarthi** [comment]
How to enable horizontal scrolling in MDCChipField?

**Comment by hardikamal** [comment]
@romoore Any update on this?

**Comment by romoore** [comment]
Hi @hardikamal. I'm no longer on the Material iOS core team, but I believe this was intended as a research investigation. I think Chips integration with TextFields might be handled by [TextControls](https://github.com/material-components/material-components-ios/tree/develop/components/TextControls/src), but @jverkoey or @andrewoverton might know better.

**Comment by andrewoverton** [comment]
Thanks @romoore.

Hi @hardikamal, the input chip control that came from this spike is still sitting in an experimental directory. Productionizing it is considered a low priority right now, unfortunately.
